resource/config.go

```diff
@@ -175,24 +176,30 @@ func (assoc AssociatedResourceConfig) MarshalJSON() ([]byte, error) {
 // Equals checks if the two configs are deeply equal to each other. Validation
 // related fields and implicit dependencies will be ignored.
 func (conf Config) Equals(other Config) bool {
+	for association, assocCfg := range conf.AssociatedAttributes {
+		otherAssocCfg, ok := other.AssociatedAttributes[association]
+		if !ok || !assocCfg.Equals(otherAssocCfg) {
+			return false
+		}
+	}
+
 	// These `Config` objects are copies. Changing the members here for equality checking does not
 	// impact the original versions.
 	conf.alreadyValidated = false
 	conf.ImplicitDependsOn = nil
 	conf.cachedImplicitDeps = nil
 	conf.cachedErr = nil
+	conf.ConvertedAttributes = nil
+	conf.AssociatedResourceConfigs = nil
+	conf.AssociatedAttributes = nil
```

Comment:
A few things happening here.  First we can nil out converted attributes since now we dont need to special case the data manager.  Second, AssociatedResourceConfigs are now taken care of with the AssociatedAttributes field, which is checked using a custom equality function at the top of the function.  So we can nil out the AssociatedAttributes field as well.

This does prompt me to wonder if theres still a point to keeping the AssociatedResourceConfigs field around. I'm not sure what other work it is doing but I thought I would raise the question

Reply:
Sounds good I think leaving as is makes sense then.  Going to close this comment